kafka/producer/base.py

```diff
@@ -188,7 +211,12 @@ def _send_messages(self, topic, partition, *msg, **kwargs):
 
         if self.async:
             for m in msg:
-                self.queue.put((TopicAndPartition(topic, partition), m, key))
+                try:
+                    self.queue.put((TopicAndPartition(topic, partition), m, key), block=True,
+                                   timeout=self.batch_send_queue_max_wait)
+                except Full:
+                    log.exception('Queue full, failed to put message "{0}" in the async queue after {1} ms'.format(
+                        m, self.batch_send_queue_max_wait))
```

Comment:
In fact the timeout on the message insertion in the async queue is in seconds.
Only the backoff for retry is in milliseconds.